bijection-util/src/main/scala/com/twitter/bijection/twitter_util/JavaFutureToTwitterFutureConverter.scala

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.bijection.twitter_util
+
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.{ Future => JFuture }
+
+import com.twitter.util.{ Future, Promise, Try }
+
+import scala.annotation.tailrec
+
+/**
+ * Utility class for converting Java futures to Twitter's
+ *
+ * @param waitTimeMs Time spent sleeping by the thread converting java futures to
+ *                   twitter futures when there are no futures to convert in ms
+ */
+private[twitter_util] class JavaFutureToTwitterFutureConverter(waitTimeMs: Long = 1000L) {
```

Comment:
@BenFradet I think we should still use the implicit parameter.

Yes, it would be wasteful.  That's why I also proposed the Timer approach, which I think is quite similar to your approach, and is appropriate for cases where there may be many futures.  The overhead will be mostly in memory for threads (if you have many concurrent futures you're waiting for) and in context switches (if you have many futures finishing at the same time).

A timer approach is almost always going to be relatively high latency, because they all work using tick mechanisms.  When you block, you move the work to the scheduler, which has the ability to work up a specific thread when there's work to do.  This means that a thread can be notified immediately if we're blocking, but if we're waiting for the next tick to check, it will add the latency until the next tick necessarily.  This will move the latency of discovering a future has finished from ~microseconds to ~milliseconds.